enhancements/operator-sdk/downstream-operator-sdk.md

```diff
@@ -0,0 +1,281 @@
+---
+title: downstream-operator-sdk
+authors:
+  - "@jmrodri"
+reviewers:
+  - "@joelanford"
+  - "@asmacdo"
+  - "@fabianvf"
+  - "@camilamacedo86"
+  - "@shawn-hurley"
+  - "@gallettilance"
+  - "@kevinrizza"
+approvers:
+  - "@joelanford"
+  - "@asmacdo"
+  - "@fabianvf"
+creation-date: 2020-09-10
+last-updated: 2020-09-10
+status: implementable
+see-also:
+replaces:
+superseded-by:
+
+---
+
+# Downstream Operator SDK
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The Operator SDK currently ships the Ansible and Helm Operator base images downstream. The purpose of this enhancement is to add a supported Operator SDK base image as well as an operator-sdk binary.
+
+## Motivation
+
+Deliver a properly supported Operator SDK downstream to support OpenShift developers.
+
+### Goals
+
+To deliver an Operator SDK base image that can be used by operator developers. Deliver an operator-sdk binary similar to how oc is delivered to customers. Continue to deliver the Ansible and Helm operator base image.
+
+### Non-Goals
+
+We will not deliver an RPM based operator-sdk package as part of this.
+
+## Proposal
+
+### User Stories [optional]
+
+* As an internal Red Hat operator developer, I would want the SDK that I use to support the supported OpenShift clusters.
+* As an internal Red Hat Ansible operator developer, I would want to have a released downstream Ansible operator image I could base my operator on.
+* As an internal Red Hat Helm operator developer, I would want to have a released downstream Ansible operator image I could base my operator on.
+* As an internal Red Hat operator developer, I would want a supported Operator SDK base image to use to build my own operators.
+* As a Red Hat Ansible operator customer, I would want to have a released downstream Ansible operator image I could base my operator on.
+* As a Red Hat Helm operator customer, I would want to have a released downstream Helm operator image I could base my operator on.
+* As a Red Hat operator customer, I would want to have a released downstream Operator SDK base image to use to build my own operators.
+* As a Red Hat operator customer, I would want a supported operator-sdk binary.
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### Repos
+
+Mirror the upstream Operator SDK repos downstream in the openshift GitHub organization. We will have a one to one upstream to downstream repo.
+
+* operator-sdk - github.com/openshift/operator-sdk
+* operator-lib - github.com/openshift/operator-lib
+* ansible-operator (future) - github/openshift/ansible-operator
```

Comment:
What do you mean by `(future)` here?

Reply:
Ansible-based operators will eventually be moved to a separate upstream repo, this is capturing that the helm/ansible operators will also need a separate downstream repo.

Reply:
Can we come back and add this when we do that work?

Reply:
Just to share it is related to the idea: https://github.com/operator-framework/enhancements/pull/44
I think another option could just add how it will. be now as @shawn-hurley described and then maybe add as a noT how we think that it could be in the future.

Reply:
@shawn-hurley are you just saying do not put it into the enhancement as future? Just leave it out and add back in when that work is planned?

Reply:
Yeah, someone reading this, then has to go find if that work was completed or not right?

Reply:
Removed.